File: sfas-ui/src/app/core/services/fund.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Fund } from '../models/fund.model';
import { environment } from '../../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class FundService {
  private http = inject(HttpClient);
  private baseUrl = environment.apiUrl + '/api/v1/funds';

  getAllFunds(): Observable<Fund[]> {
    return this.http.get<Fund[]>(this.baseUrl);
  }

  createFund(fund: Partial<Fund>): Observable<Fund> {
    return this.http.post<Fund>(this.baseUrl, fund);
  }

  updateFundStatus(id: string, status: 'APPROVED' | 'REJECTED' | 'PENDING', comments?: string): Observable<Fund> {
    return this.http.patch<Fund>(`${this.baseUrl}/${id}/status`, { status, comments });
  }
}

File: sfas_mobile/lib/features/funds/services/fund_service.dart
import 'package:dio/dio.dart';
import '../../../core/constants/endpoints.dart';
import '../models/fund.dart';

class FundService {
  final Dio _dio;

  FundService(this._dio);

  Future<List<Fund>> getFunds() async {
    try {
      final response = await _dio.get(Endpoints.funds);
      final List<dynamic> data = response.data;
      return data.map((json) => Fund.fromJson(json)).toList();
    } catch (e) {
      rethrow;
    }
  }

  Future<Fund> createFund(Fund fund) async {
    try {
      final response = await _dio.post(
        Endpoints.funds,
        data: fund.toJson(),
      );
      return Fund.fromJson(response.data);
    } catch (e) {
      rethrow;
    }
  }

  Future<Fund> updateFundStatus(String id, String status) async {
    try {
      final response = await _dio.patch(
        Endpoints.updateFundStatus(id),
        queryParameters: {'status': status},
      );
      return Fund.fromJson(response.data);
    } catch (e) {
      rethrow;
    }
  }
}
